tests/pf/test_forward_Mag_Linear.py

```diff
@@ -101,6 +105,116 @@ def get_block_inds(grid, block):
+    hzind = [(cs, 41)]
+    mesh = discretize.TensorMesh([hxind, hyind, hzind], "CCC")
+
+    # create a model of two blocks, 1 inside the other
+    block1 = np.array([[-1.5, 1.5], [-1.5, 1.5], [-1.5, 1.5]])
+    block2 = np.array([[-0.7, 0.7], [-0.7, 0.7], [-0.7, 0.7]])
+
+    def get_block_inds(grid, block):
+        return np.where(
+            (grid[:, 0] > block[0, 0])
+            & (grid[:, 0] < block[0, 1])
+            & (grid[:, 1] > block[1, 0])
+            & (grid[:, 1] < block[1, 1])
+            & (grid[:, 2] > block[2, 0])
+            & (grid[:, 2] < block[2, 1])
+        )
+
+    block1_inds = get_block_inds(mesh.cell_centers, block1)
+    block2_inds = get_block_inds(mesh.cell_centers, block2)
+
+    model = np.zeros(mesh.n_cells)
+    model[block1_inds] = chi1
+    model[block2_inds] = chi2
+
+    active_cells = model != 0.0
+    model_reduced = model[active_cells]
+
+    # Create reduced identity map for Linear Problem
+    idenMap = maps.IdentityMap(nP=int(sum(active_cells)))
+
+    # Create plane of observations
+    xr = np.linspace(-20, 20, nx)
+    dxr = xr[1] - xr[0]
+    yr = np.linspace(-20, 20, ny)
+    dyr = yr[1] - yr[0]
+    X, Y = np.meshgrid(xr, yr)
+    Z = np.ones_like(X) * 3.0
+    locXyz = np.c_[X.reshape(-1), Y.reshape(-1), Z.reshape(-1)]
+    components = ["tmi", "tmi_x", "tmi_y", "tmi_z"]
+
+    rxLoc = mag.Point(locXyz, components=components)
+    srcField = mag.UniformBackgroundField(
+        [rxLoc], amplitude=H0[0], inclination=H0[1], declination=H0[2]
+    )
+    survey = mag.Survey(srcField)
+
+    # Creat reduced identity map for Linear Problem
+    idenMap = maps.IdentityMap(nP=int(sum(active_cells)))
+
+    sim = mag.Simulation3DIntegral(
+        mesh,
+        survey=survey,
+        chiMap=idenMap,
+        ind_active=active_cells,
+        store_sensitivities="forward_only",
+        n_processes=None,
+    )
+
+    data = sim.dpred(model_reduced)
+    tmi = data[0::4]
+    d_x = data[1::4]
+    d_y = data[2::4]
+    d_z = data[3::4]
+
+    # Compute analytical response from magnetic prism
+    prism_1 = MagneticPrism(block1[:, 0], block1[:, 1], chi1 * b0 / mu_0)
+    prism_2 = MagneticPrism(block2[:, 0], block2[:, 1], -chi1 * b0 / mu_0)
+    prism_3 = MagneticPrism(block2[:, 0], block2[:, 1], chi2 * b0 / mu_0)
+
+    d = (
+        prism_1.magnetic_field_gradient(locXyz)
+        + prism_2.magnetic_field_gradient(locXyz)
+        + prism_3.magnetic_field_gradient(locXyz)
+    ) * mu_0
+    tmi_x = (d[:, 0, 0] * b0[0] + d[:, 0, 1] * b0[1] + d[:, 0, 2] * b0[2]) / H0[0]
+    tmi_y = (d[:, 1, 0] * b0[0] + d[:, 1, 1] * b0[1] + d[:, 1, 2] * b0[2]) / H0[0]
+    tmi_z = (d[:, 2, 0] * b0[0] + d[:, 2, 1] * b0[1] + d[:, 2, 2] * b0[2]) / H0[0]
+    np.testing.assert_allclose(d_x, tmi_x, rtol=1e-10, atol=1e-12)
+    np.testing.assert_allclose(d_y, tmi_y, rtol=1e-10, atol=1e-12)
+    np.testing.assert_allclose(d_z, tmi_z, rtol=1e-10, atol=1e-12)
+
+    # finite difference test y-grad
```

Comment:
@santisoler : I did add a finite difference test here.
I had to increase the number of receiver to obtain an ok response.
This is similar to what you propose (but instead of one global receiver grid, you have several). I think having a single  grid make the test syntax less cumbersome and easier to debug later on.

The test is still very fast to run with the increased number of receivers (<3 s on my machine).

See if that solution works for you.

Reply:
Sounds good! I think we can merge this now 🚀

Reply:
~5s on Azure
![image (92)](https://github.com/simpeg/simpeg/assets/97514898/a4ab5992-baae-4e8e-acde-c3b7add199e5)